stac_generator/raster_data/generator.py

```diff
@@ -0,0 +1,171 @@
+                geometry = Polygon(
+                    [
+                        (bounds.left, bounds.bottom),
+                        (bounds.left, bounds.top),
+                        (bounds.right, bounds.top),
+                        (bounds.right, bounds.bottom),
+                        (bounds.left, bounds.bottom),
+                    ]
+                )
+                geometry_geojson = mapping(geometry)
+
+                if source_cfg.datetime is None:
+                    raise ValueError("source_cfg.datetime cannot be None")
+
+                # Ensure datetime is timezone-aware
+                if source_cfg.datetime.tzinfo is None:
+                    datetime_aware = source_cfg.datetime.replace(tzinfo=UTC)
+                else:
+                    datetime_aware = source_cfg.datetime
+
+                # Create STAC Item
+                item = pystac.Item(
+                    id=source_cfg.prefix,
+                    geometry=geometry_geojson,
+                    bbox=bbox,
+                    datetime=datetime_aware,
+                    properties={
+                        "eo:snow_cover": 0,
+                        "eo:cloud_cover": 0,
+                        "proj:epsg": crs.to_epsg(),
+                        "proj:shape": shape,
+                        "proj:transform": list(transform)[:9],
+                    },
+                )
+
+                # Initialize extensions
+                EOExtension.ext(item, add_if_missing=True)
+                proj_ext = ProjectionExtension.ext(item, add_if_missing=True)
+                proj_ext.apply(epsg=crs.to_epsg(), shape=shape, transform=list(transform)[:9])
+
+                # Create EO and Raster bands
+                eo_bands = []
+                raster_bands = []
+                for idx, band_info in enumerate(source_cfg.bands, start=1):
+                    eo_band = Band.create(
+                        name=f"B{idx}",
+                        common_name=BAND_MAPPING.get(band_info.name.lower(), None),
+                        center_wavelength=band_info.wavelength,
+                        description=f"Common name: {BAND_MAPPING.get(band_info.name.lower(), 'unknown')}",
+                    )
+                    eo_bands.append(eo_band)
+
+                    raster_band = RasterBand.create(
+                        nodata=band_info.nodata or 0,
+                        data_type=DataType(band_info.data_type or "uint16"),
+                    )
+                    raster_bands.append(raster_band)
+
+                item.properties["eo:bands"] = [band.to_dict() for band in eo_bands]
+
+                # Create Asset and Add to Item
+                asset = pystac.Asset(
+                    href=source_cfg.location,
+                    media_type=pystac.MediaType.GEOTIFF,
+                    roles=["data"],
+                    title="Raster Data",
+                )
+                item.add_asset("image", asset)
```

Comment:
For consistency, let's use "data" as asset name since this was used for both the vector and point generator. This is for the stac loader.